### ISSUE TITLE
Blue channel of anisotropyTexture causing odd behavior

### ISSUE THREAD

**Issue by echadwick-artist** [mrdoob/three.js#26694]
### Description

When the blue channel of an anisotropyTexture is set to zero, three.js seems to exhibit odd rendering behavior.
![2023-09-04 18_55_47-NVIDIA GeForce Overlay DT](https://github.com/mrdoob/three.js/assets/40177036/03122d01-c783-4d6b-bd9e-ec2f8be7d72f)

Maybe related to #25701 ?

anisotropyTexture looks like this. Both spheres are assigned a vertical direction (.5,1), left sphere has full aniso (blue 1.0), right sphere has zero (blue 0))
![anisotropyTexture](https://github.com/mrdoob/three.js/assets/40177036/0539bd83-a12a-4fa8-b018-5fb38badc751)

Other renderers:
![2023-09-04 18_53_04-NVIDIA GeForce Overlay DT](https://github.com/mrdoob/three.js/assets/40177036/632c9f8c-1a00-4e24-acd0-d2ef74378862)

![2023-09-04 18_53_58-NVIDIA GeForce Overlay DT](https://github.com/mrdoob/three.js/assets/40177036/a1fe398b-3035-4f99-b540-160c97a3abe0)

This issue can also be seen in the glTF Sample Asset [AnisotropyBarnLamp](https://github.com/KhronosGroup/glTF-Sample-Assets/tree/main/Models/AnisotropyBarnLamp#anisotropy-barn-lamp). Notice the shading differences on the flat knob at the top, knurled ring grip at middle, and 3 shade screws at the bottom. Viewers both use the IBL [Wide Street 01 from Poly Haven](https://polyhaven.com/a/wide_street_01).
![2023-09-04 22_31_50-NVIDIA GeForce Overlay DT](https://github.com/mrdoob/three.js/assets/40177036/dceaa838-082a-49d6-806a-d0540bf724d1)

anisotropyBarnLamp's anisotropyStrength (blue channel):
![2023-09-04 22_33_02-NVIDIA GeForce Overlay DT](https://github.com/mrdoob/three.js/assets/40177036/4d4c72fa-ccbc-463c-84f6-482998db22e1)

### Reproduction steps

1. Go to https://threejs.org/editor/
2. Drag-n-drop attached glTF.
3. Set the Environment to MODELVIEWER.

### Code

glTF material:
```
    {
      "name": "Copper",
      "pbrMetallicRoughness": {
        "baseColorFactor": [
          0.8,
          0.25,
          0.1,
          1.0
        ],
        "metallicFactor": 1,
        "roughnessFactor": 0.2
      },
      "normalTexture": {
        "index": 0
      },
      "extensions": {
        "KHR_materials_anisotropy": {
          "anisotropyStrength": 1.0,
          "anisotropyTexture": {
            "index": 1
          }
        }
      }
    }
```

### Live example

Sample model using [KHR_materials_anisotropy](https://github.com/KhronosGroup/glTF/tree/main/extensions/2.0/Khronos/KHR_materials_anisotropy). Both spheres are in a single mesh, using same material, and the normalTexture is flat 128,128,255.
[anisotropyTexture-v-Metalness.zip](https://github.com/mrdoob/three.js/files/12518019/anisotropyTexture-v-Metalness.zip)

### Version

r156

### Device

Desktop

### Browser

Chrome, Firefox

### OS

Windows

**Comment by GitHubDragonFly** [comment]
I would have a simple question related to the ShaderLib.js file and its Anisotropy declarations:

```js
	anisotropyVector: { value: /*@__PURE__*/ new Vector2() },
	anisotropyMap: { value: null },
	anisotropyMapTransform: { value: /*@__PURE__*/ new Matrix3() },
```

Are these actually correct, specifically the `anisotropyVector`?

Should there be `anisotropy` as used in three.js (or as the specs call it anisotropyStrength) and possibly `anisotropyRotation`?

Not really an expert in these but am just using a common sense logic and comparing these to let's say the `clearcoat` declaration of `clearcoat: { value: 0 }`.

**Comment by Mugen87** [comment]
`anisotropyVector` is computed based on `anisotropy` and `anisotropyRotation`. Only `anisotropyVector` exists as a uniform in the shader.

**Comment by echadwick-artist** [comment]
Shouldn’t it be using Matrix2 instead? This is a 2d vector stored only in red and green. The blue channel needs to be omitted from the vector.

**Comment by elalish** [comment]
No, that transform is applied to the UVs, not the vector - it's Matrix3 because it's an affine transform for 2D; same reason we use Matrix4 on 3D vectors.

**Comment by elalish** [comment]
The lamp difference might be related to #26967, which I need to look into. Regarding your spheres, can you double check your model? I don't see your blue/green texture in there, and it also has a uniform normal map texture? Also, both spheres are in a single mesh, which makes it a bit hard to see what's happening. Do you get different results when you set the strength factor to zero?

**Comment by WestLangley** [comment]
@elalish Tracking this down, one problem is this line:

https://github.com/mrdoob/three.js/blob/bf7d99e30e2393e5f7bcdc9b6781406ffca58374/src/renderers/shaders/ShaderChunk/lights_physical_fragment.glsl.js#L127

The blue channel of the texture cannot be zero in the three.js implementation, although zero is a valid value according to the spec.

**Comment by elalish** [comment]
Thanks, I just noticed that too. However, I think there's a bigger problem. I removed all the textures besides anisotropy from the lamp, and it appears that when anisotropy strength => 0, then decreasing roughness darkens the result (loses specular energy?):
<img width="1548" alt="image" src="https://github.com/mrdoob/three.js/assets/1649964/cc76f3c2-c09f-4643-adaa-d9a216b239ca">
I need to look at those D and V terms I got from Filament; I probably messed something up, unless there's an energy conservation term I didn't realize was having an effect somewhere.
